### ISSUE TITLE
NodePort services seems to not work in latest version

### ISSUE THREAD

**Issue by ventsislav-georgiev** [abiosoft/colima#410]
### Description

The k8s services of type NodePort no longer listen on localhost. In v0.4.4 they work fine.

### Version

Colima Version: HEAD-6c4889b
Lima Version: 0.11.3
Qemu Version: 7.0.0

### Operating System

- [ ] macOS Intel
- [X] macOS M1
- [ ] Linux

### Reproduction Steps

1. `colima start --with-kubernetes`
2. `kubectl create deployment hello --image=gcr.io/google-samples/hello-app:1.0`
3. `kubectl expose deploy hello --type NodePort --port 80 --target-port 8080`
4. `curl localhost:$(kubectl get svc hello -o go-template='{{range.spec.ports}}{{.nodePort}}{{end}}')`

**Comment by abiosoft** [comment]
Is this still the case?

Thanks for reporting.

**Comment by jdambly-ns** [comment]
I was having similar issues, but it's working for me

```
(⎈ |colima:default)➜  redis-operator git:(master) curl localhost:$(kubectl get svc hello -o go-template='{{range.spec.ports}}{{.nodePort}}{{end}}')
Hello, world!
Version: 1.0.0
Hostname: hello-6d64d54cc8-sgzgl
(⎈ |colima:default)➜  redis-operator git:(master) curl localhost:$(kubectl get svc rfs-redisfailover -o go-template='{{range.spec.ports}}{{.nodePort}}{{end}}')
curl: (3) URL using bad/illegal format or missing URL
400 Bad Request: malformed Host header%
(⎈ |colima:default)➜  redis-operator git:(master)
```

I'm on a macOS with M1, minikube doesn't really work for me any more b/c it only supports the docker driver which doesn't let me expose nodePorts, so colima is good replacement. I didn't realize that nodePort will bind to local host

is hostPort supported as well?

**Comment by abiosoft** [comment]
I'll verify this. Nothing has changed (code-wise) that should cause the change in behaviour.

**Comment by abiosoft** [comment]
I can confirm this behaviour. However, I am not sure of the cause.

If the goal is mainly for port accessibility on the host, `NodePort` still works if network address is enabled. And `LoadBalancer` works regardless.

**Comment by jandubois** [comment]
> If k3s's version is newer than `v1.23.7+k3s1`, this issue will be occur.

It is a change in upstream kubernetes and affects all newer releases. I have a comprehensive list in https://github.com/rancher-sandbox/rancher-desktop/issues/2593#issuecomment-1192278825

There is some dedicated agent code in RD that deals with this, but we still need to backport it to Lima: https://github.com/lima-vm/lima/issues/1021#issuecomment-1231876644

> If the goal is mainly for port accessibility on the host, `NodePort` still works if network address is enabled. And `LoadBalancer` works regardless.

Yes, it only affects the automatic port forwarding.

**Comment by ventsislav-georgiev** [comment]
Thanks @mbcoguno and @jandubois! I've currently reverted to v1.23.6+k3s1 to keep having NodePort auto-forwarding.
@abiosoft should we keep this issue open?

**Comment by abiosoft** [comment]
> @abiosoft should we keep this issue open?

I would say yes to keep track since a fix upstream in Lima is planned.

**Comment by abiosoft** [comment]
This seems to be resolved with Lima v0.15.0.
